### PR TITLE
Time and timestamp utility refactoring

### PR DESCRIPTION
**Pull request by thehenrytsai**
1. Exported all time utilities for downstream external usage
1. Renamed `getCurrentTimeInHighPrecision()` -> `getCurrentTimestamp()` for dev friendliness
1. Removed redundant `createDateString()`
1. Grouped all time utilities into a class
1. Replaced all direct usage of temporal library

### CODE REVIEW

File: src/utils/time.ts
Comment (thehenrytsai) [inline src/utils/time.ts:1]: No actual change, just opportunistically reordering/grouping similar methods together.

File: tests/handlers/records-query.spec.ts [hunk 1]
```diff
@@ -1608,7 +1605,7 @@ export function testRecordsQueryHandler(): void {
       });
 
       it('should return 400 if published is set to false and a datePublished range is provided', async () => {
-        const firstDayOf2021 = createDateString(new Date(2021, 1, 1));
+        const firstDayOf2021 = Time.getCurrentTimestamp();
```
Comment (LiranCohen) [inline tests/handlers/records-query.spec.ts:1608]: nit: rename `firstDayOf2021` to something like `startDate`

**Review by LiranCohen — DISMISSED** [review]
This is great! Nice cleanup effort! 🎸  found one nit but we can just merge this.

File: tests/handlers/records-query.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/handlers/records-query.spec.ts:1608]: Thanks for picking up! Fixing!

**Review by LiranCohen — APPROVED** [review]

**Pull request merged**